lib/bloc/pos_catalog/bloc.dart

```diff
@@ -276,8 +293,130 @@ class PosCatalogBloc with AsyncActionsHandler {
     _loadItems();
   }
 
+  void _loadSelectedReportTimeRange() async {
+    final prefs = await ServiceInjector().sharedPreferences;
+    PosReportTimeRange timeRange;
+    if (prefs.containsKey(_kPosReportTimeRangeKey)) {
+      final intRepresentation = prefs.getInt(_kPosReportTimeRangeKey);
+      switch (intRepresentation) {
+        case _kPosReportTimeRangeDaily:
+          timeRange = PosReportTimeRange.daily();
+          break;
+        case _kPosReportTimeRangeWeekly:
+          timeRange = PosReportTimeRange.weekly();
+          break;
+        case _kPosReportTimeRangeMonthly:
+          timeRange = PosReportTimeRange.monthly();
+          break;
+        case _kPosReportTimeRangeCustom:
+          final fallback = DateTime.now().millisecondsSinceEpoch;
+          final startDate = DateTime.fromMillisecondsSinceEpoch(
+            prefs.getInt(_kPosReportTimeRangeStartKey) ?? fallback,
+          );
+          final endDate = DateTime.fromMillisecondsSinceEpoch(
+            prefs.getInt(_kPosReportTimeRangeEndKey) ?? fallback,
+          );
+          timeRange = PosReportTimeRange.custom(startDate, endDate);
+          break;
+        default:
+          timeRange = PosReportTimeRange.daily();
+          break;
+      }
+    } else {
+      timeRange = PosReportTimeRange.daily();
+    }
+    _posReportRange.add(timeRange);
+  }
+
+  Future _updatePosReportTimeRange(
+    UpdatePosReportTimeRange action,
+  ) async {
+    final prefs = await ServiceInjector().sharedPreferences;
+
+    var intRepresentation = 0;
+    final timeRange = action.range;
+    if (timeRange is PosReportTimeRangeDaily) {
+      intRepresentation = _kPosReportTimeRangeDaily;
+    } else if (timeRange is PosReportTimeRangeWeekly) {
+      intRepresentation = _kPosReportTimeRangeWeekly;
+    } else if (timeRange is PosReportTimeRangeMonthly) {
+      intRepresentation = _kPosReportTimeRangeMonthly;
+    } else if (timeRange is PosReportTimeRangeCustom) {
+      intRepresentation = _kPosReportTimeRangeCustom;
+    }
+
+    prefs.setInt(_kPosReportTimeRangeKey, intRepresentation);
+    prefs.setInt(
+      _kPosReportTimeRangeStartKey,
+      timeRange.startDate.millisecondsSinceEpoch,
+    );
+    prefs.setInt(
+      _kPosReportTimeRangeEndKey,
+      timeRange.startDate.millisecondsSinceEpoch,
+    );
+    _posReportRange.add(timeRange);
+    _loadItems();
+  }
+
+  void _loadSelectedReportResult() async {
+    _posReportResult.add(PosReportResult.empty());
+  }
+
+  Future _fetchPosReport(
+    FetchPosReport action,
+  ) async {
+    _posReportResult.add(PosReportResult.load());
+    final timeRange = action.range;
+    final payments = await _breezLib.getPayments();
```

Comment:
Why don't we query the db for sales items directly? Is there a reason we go through the payments list?

Reply:
The problem is the time range, I did not find a way to fetch the sales by date.
If you see a way to fetch it using this time range filter it would be much better than reading the payment first.

Reply:
Indeed you are right. The sale_payments table should have a payment_time column...
Maybe it makes sense to add a migration step to add this time to all existing payments? And then change the implementation to use pure SQL query?

Reply:
@kingonly The CSV uses the payment list too.

@roeierez I'm checking this migration thing

Reply:
@ademar111190  great!
In our sqflite plugin we can pass onUpgrade parameter to the openDatabase function for migration to run.

Reply:
@roeierez I did add a migration process to SQL, please take a double care on this part because it is hard to test and I'm afraid of something wrong that I did not catch 🙏